components/homme/src/share/cxx/KernelVariables.hpp

```diff
@@ -107,6 +128,7 @@ struct KernelVariables {
 
   int ie, iq;
   const int team_idx;
+  TeamUtils<ExecSpace>* team_utils;
```

Comment:
Do you really need to store team_utils inside the kv object? It seems to me that it is superfluous.

Reply:
I think we do if we want to keep the current approach. I wanted to integrate TeamUtils with KernelVariables since there's overlap in functionality and to minimize changes to functors. In particular, leveraging the KernelVariables destructor to release workspace indices would not be possible without this member.